### ISSUE TITLE
[Chip] Chip icon does not support icons with different width and height

### ISSUE THREAD

**Issue by rafaelsm** [material-components/material-components-android#1241]
**Description:** When using a drawable with width greater than height, vice-versa, the icon became stretched:
<img width="201" alt="Screen Shot 2020-04-23 at 12 32 49" src="https://user-images.githubusercontent.com/893592/80120545-8b15b900-8561-11ea-999a-f775801c1e19.png">

Even after use the attribute chipIconSize:
<img width="189" alt="Screen Shot 2020-04-23 at 12 55 42" src="https://user-images.githubusercontent.com/893592/80120756-d62fcc00-8561-11ea-9e38-21c001716a0c.png">

**Expected behavior:** Should show the image correctly:
<img width="213" alt="Screen Shot 2020-04-23 at 12 33 18" src="https://user-images.githubusercontent.com/893592/80120820-ee075000-8561-11ea-82b6-a346559949f2.png">

**Source code:** The code snippet which is causing this issue. Please consider attaching a minimal sample app that reproduces the issue.
`<com.google.android.material.chip.Chip
      android:id="@+id/cat_chip_filter_chip"
      style="@style/Widget.MaterialComponents.Chip.Filter"
      android:layout_width="wrap_content"
      android:layout_height="wrap_content"
      android:layout_marginTop="8dp"
      android:layout_marginBottom="8dp"
      android:layout_marginStart="16dp"
      android:layout_marginEnd="16dp"
      android:layout_marginLeft="16dp"
      android:layout_marginRight="16dp"
      app:chipIcon="@drawable/ic_stars_4"
      app:chipIconVisible="true"
      android:layout_below="@id/cat_chip_entry_chip"
      android:layout_gravity="center_vertical|start"
      android:layout_toEndOf="@id/cat_chip_filter"
      android:layout_toRightOf="@id/cat_chip_filter"
      android:ellipsize="end"
      android:text="@string/cat_chip_text"/>`

The drawable xml:
`<vector xmlns:android="http://schemas.android.com/apk/res/android"
    android:width="70dp"
    android:height="14dp"
    android:viewportWidth="70"
    android:viewportHeight="14">
  <path
      android:pathData="M66.0332,13.6831C66.1734,13.6869 66.3085,13.6316 66.4032,13.5316C66.4979,13.4315 66.5429,13.2966 66.5266,13.1618L66.0079,8.8477L69.0493,5.6518C69.1669,5.528 69.2065,5.3525 69.1531,5.1925C69.0997,5.0325 68.9615,4.9126 68.7912,4.8786L64.386,4.0221L62.1847,0.2365C62.0989,0.0903 61.9385,0 61.7647,0C61.5909,0 61.4305,0.0903 61.3447,0.2365L59.1434,4.0221L54.7382,4.8786C54.5679,4.9126 54.4297,5.0325 54.3763,5.1925C54.3229,5.3525 54.3625,5.528 54.4801,5.6518L57.5215,8.8477L57.0028,13.1618C56.9833,13.3291 57.0583,13.4936 57.1992,13.5924C57.3401,13.6912 57.5251,13.709 57.6834,13.639L61.7647,11.8306L65.846,13.639C65.9047,13.6657 65.9684,13.6807 66.0332,13.6831Z"
      android:strokeWidth="1"
      android:fillColor="#FED200"
      android:fillType="evenOdd"
      android:strokeColor="#00000000"/>
  <path
      android:pathData="M47.9156,13.6831C48.0558,13.6869 48.1908,13.6316 48.2855,13.5316C48.3802,13.4315 48.4253,13.2966 48.409,13.1618L47.8903,8.8477L50.9316,5.6518C51.0492,5.528 51.0889,5.3525 51.0355,5.1925C50.9821,5.0325 50.8438,4.9126 50.6736,4.8786L46.2684,4.0221L44.0671,0.2365C43.9813,0.0903 43.8209,0 43.6471,0C43.4732,0 43.3128,0.0903 43.227,0.2365L41.0257,4.0221L36.6206,4.8786C36.4503,4.9126 36.3121,5.0325 36.2587,5.1925C36.2052,5.3525 36.2449,5.528 36.3625,5.6518L39.4038,8.8477L38.8851,13.1618C38.8656,13.3291 38.9407,13.4936 39.0816,13.5924C39.2225,13.6912 39.4075,13.709 39.5658,13.639L43.6471,11.8306L47.7283,13.639C47.7871,13.6657 47.8507,13.6807 47.9156,13.6831Z"
      android:strokeWidth="1"
      android:fillColor="#FED200"
      android:fillType="evenOdd"
      android:strokeColor="#00000000"/>
  <path
      android:pathData="M29.7979,13.6831C29.9382,13.6869 30.0732,13.6316 30.1679,13.5316C30.2626,13.4315 30.3077,13.2966 30.2913,13.1618L29.7726,8.8477L32.814,5.6518C32.9316,5.528 32.9712,5.3525 32.9178,5.1925C32.8644,5.0325 32.7262,4.9126 32.5559,4.8786L28.1508,4.0221L25.9494,0.2365C25.8636,0.0903 25.7032,0 25.5294,0C25.3556,0 25.1952,0.0903 25.1094,0.2365L22.9081,4.0221L18.5029,4.8786C18.3326,4.9126 18.1944,5.0325 18.141,5.1925C18.0876,5.3525 18.1273,5.528 18.2448,5.6518L21.2862,8.8477L20.7675,13.1618C20.748,13.3291 20.823,13.4936 20.9639,13.5924C21.1048,13.6912 21.2898,13.709 21.4481,13.639L25.5294,11.8306L29.6107,13.639C29.6694,13.6657 29.7331,13.6807 29.7979,13.6831Z"
      android:strokeWidth="1"
      android:fillColor="#FED200"
      android:fillType="evenOdd"
      android:strokeColor="#00000000"/>
  <path
      android:pathData="M11.6803,13.6831C11.8205,13.6869 11.9555,13.6316 12.0502,13.5316C12.145,13.4315 12.19,13.2966 12.1737,13.1618L11.655,8.8477L14.6964,5.6518C14.8139,5.528 14.8536,5.3525 14.8002,5.1925C14.7468,5.0325 14.6085,4.9126 14.4383,4.8786L10.0331,4.0221L7.8318,0.2365C7.746,0.0903 7.5856,0 7.4118,0C7.2379,0 7.0775,0.0903 6.9917,0.2365L4.7904,4.0221L0.3853,4.8786C0.215,4.9126 0.0768,5.0325 0.0234,5.1925C-0.03,5.3525 0.0096,5.528 0.1272,5.6518L3.1685,8.8477L2.6498,13.1618C2.6303,13.3291 2.7054,13.4936 2.8463,13.5924C2.9872,13.6912 3.1722,13.709 3.3305,13.639L7.4118,11.8306L11.493,13.639C11.5518,13.6657 11.6155,13.6807 11.6803,13.6831Z"
      android:strokeWidth="1"
      android:fillColor="#FED200"
      android:fillType="evenOdd"
      android:strokeColor="#00000000"/>
</vector>
`

**Android API version:** 29

**Material Library version:** 1.2.0-alpha06
**Device:** Emulator, Pixel
To help us triage faster, please check to make sure you are using the [latest version](https://github.com/material-components/material-components-android/releases) of the library.

We also happily accept [pull requests](https://github.com/material-components/material-components-android/pulls).

**Comment by vbuberen** [comment]
Can confirm the issue.

Here is my Chip:
 ```
<com.google.android.material.chip.Chip
                style="@style/Widget.MaterialComponents.Chip.Action"
                android:layout_width="wrap_content"
                android:layout_height="wrap_content"
                android:text="@string/play_all"
                android:textColor="@color/white"
                app:chipIconSize="48dp"
                app:chipIcon="@mipmap/ic_media_play"
                app:chipStrokeColor="@color/white"
                app:chipStrokeWidth="1dp"
                app:chipSurfaceColor="@color/black_transparent" />
```
And here is the result:
<img width="263" alt="Screenshot 2020-04-23 at 21 11 30" src="https://user-images.githubusercontent.com/13467769/80134555-95f33c80-85a7-11ea-98ab-68920b493eca.png">

Note that `chipMinHeight` has no `wrap_content` property, so there is no way to set custom size for icons without manually setting height for the whole chip.

Works the same way on `1.1.0` and `1.2.0-alpha06`

**Comment by wcshi** [comment]
Thank you for contributing, we are waiting for design feedback on this usage.

**Comment by wcshi** [comment]
tl;dr: 
Design is ok with the chip icon width and height not being the same IF and ONLY IF the icon height doesn't exceed 24dp and it does not not affect the chip's overall height. 

Basically, **Material supports variable chip icon width with fixed height**.

Full feedback from Design:

The intended use of chip icon is for identifying entities like an avatar, logo, or icon. The example in the Github is using four stars as a single icon, but that style of icon is more of content than it is an identifying marker.

The dimensions we care about most are the icon height and the padding around it. If the width needs to be variable to support diff. types of graphics, that seems OK.

We don't want the max chip icon height to exceed 24dp to maintain consistent padding within the container to avoid getting too close to any edge.

CAUTION: Increasing the size of the chip really depends on the use case, like the design it sits within and definitely a caution. If it is too large it might overwhelm the hierarchy amongst other components but an increase in size doesn't hurt as far as a11y usability is concerned.

**Comment by wcshi** [comment]
@rafaelsm could you please update your PR to add support for setting chip icon width only?

**Comment by rafaelsm** [comment]
Ok, I'll make the change to support only width

**Comment by vbuberen** [comment]
@wcshi Thanks for your reply.
What about my use case? I just want to increase height to match the icon as it is provided by app designers.

**Comment by rafaelsm** [comment]
@wcshi I've made the change, now it's considering only the drawable width.

**Comment by vbuberen** [comment]
Also, it is so strange to have iconSize attribute and have such strange justification for element being not properly resizable.

**Comment by wcshi** [comment]
@rafaelsm 
I'm curious to see what happens when a chip with a wide leading icon is checked, does it whole chip width suddenly jump shrink? Are you expecting people to provide an identical size checked icon?

**Comment by wcshi** [comment]
@vbuberen your icon looks like it has the same height/width, did you try setting both the chipIconSize and the chip min height?

**Comment by rafaelsm** [comment]
> @rafaelsm
> I'm curious to see what happens when a chip with a wide leading icon is checked, does it whole chip width suddenly jump shrink? Are you expecting people to provide an identical size checked icon?

Well in fact, when I faced this problem my case was that I needed to show a range of stars 1~5 and when check the chip only the background color so there's no scale problem. 
In the case of the check icon, should be provided an icon of the same size so the chip won't shrink/stretch, or disable the chipIconUseDrawableSize.

**Comment by vbuberen** [comment]
@wcshi I asked about my case, because it is not exactly what your feedback from design team addresses. The issues is that wrapping simply doesn't work. Like here is what happens when I set both `chipIconSize` and `chipMinHeight` to 48dp (as original icon size):
<img width="231" alt="Screenshot 2020-04-27 at 17 08 09" src="https://user-images.githubusercontent.com/13467769/80381595-d30e4600-88a9-11ea-8a9b-68a934087b1a.png">
As you might see it doesn't respect size changes, so I have odd paddings there. 
Moreover, I can' just make Chip bigger, but with the original aspect ratio (like the original chip, but bigger) except be setting `scaleX` and `scaleY`. Finally, even is I could achieve size I need `textSize` is just ignored and I am ending up with big chip and small text.

**Comment by wcshi** [comment]
@vbuberen what do you want the button with your giant play icon to look like?
The odd padding is due to the fact that your icon is taller than the recommended 24dp

**Comment by wcshi** [comment]
@vbuberen it looks like you are trying to create a "play button", in which case it may be more appropriate to use a MaterialButton.

**Comment by vbuberen** [comment]
@wcshi you still didn't get why I shared all that info above.

What I want to achieve? I want to follow the designs provided to me. Yes, Button should work for that, but Actions Chip also works for that.

If it was true about 24 dp for the icon size I highly doubt that there will be any `iconSize` or `chipMinHeight` properties for it.

Moreover, there are no mentions of that `recommendation` in the official guidelines
https://material.io/components/chips#filter-chips or  https://material.io/develop/android/components/chip/ 

So, if there are no mentions or restrictions in the official guidelines it means that the issue I have is a bug, which should be addressed. 

Additionally, the statement provided as a design feedback has nothing to consider it official since it is not in the documentation.

**Comment by wcshi** [comment]
I am trying to help you by verifying and trying to understand what you / your designer wants and you are being condescending and rude. 

You are not following the spec because the spec for outlined action chip clearly show that it should be 24dp x 24dp https://material.io/components/chips#specs

Please see the [usage section](https://material.io/components/chips#usage) for the intended uses of chip, esp. [action chip](https://material.io/components/chips#action-chips).

"Chips allow users to enter information, make selections, filter content, or trigger actions. While **buttons** are **expected to appear consistently** and with **familiar calls to action**, **chips** should **appear dynamically as a group of multiple interactive elements.** **Action chips offer actions related to primary content. They should appear dynamically and contextually in a UI.**

An alternative to action chips are buttons, which should appear persistently and consistently."

Does your "play all" button appear dynamically or is it always on the screen? 

`iconSize` is to allow devs to customize smaller icon sizes
`chipMinHeight` was adde to ensure that if a dev specifies a tiny font, the chip will still be visible / accessible and look decent.

**Comment by vbuberen** [comment]
Thanks for help, but I am trying to point you to some missed points, because I already replaced Chip with Button when I left my very first comment in this issue.

Ok, you are pointing me to specs that Chip icons should be 24dp - great, agree.

> `chipMinHeight` was adde to ensure that if a dev specifies a tiny font, the chip will still be visible / accessible and look decent.

At the same time I can't agree here, because in the same specs you shared Chip also has specified height, so how would small font influence this height? So it is either your statement about `chipMinHeight` is wrong or your statement about icon having to be 24dp is wrong. And this is what I would love to find out as well as lots of other devs, because it is just another demonstration of the opinion I shared in my previous message that guidelines aren't clear enough or they aren't followed in this library.

<img width="260" alt="Screenshot 2020-04-29 at 00 48 56" src="https://user-images.githubusercontent.com/13467769/80541268-86169680-89b3-11ea-96c8-044fca5c554c.png">

**Comment by wcshi** [comment]
Agreed guidance on Material.io could be more explicit.

For your case, iiuc, you want to keep the same proportional paddings around the leading chip icon but specify a drawable that is 48dp.
By setting both chipIconSize and minChipHeight to 48dp, you did not account for the top and bottom spacing, which is normally 8dp (32-24). But since your icon is 2x the default/recommended icon size, you need to scale up the padding as well.

By default, the action chip has:
chip start padding: 4dp
icon start padding: 0dp
icon end padding: 0dp
text start padding: 8dp 

Try the following:
chipIconSize = 48dp
chipMindHeight = chipIconSize + (8dp) * 2 = 48 + 16 = 64 dp
chip start padding: 4dp * 2 = 8dp

**Comment by wcshi** [comment]
Merged in https://github.com/material-components/material-components-android/pull/1242
Closing this bug.

**Comment by mecoFarid** [comment]
@wcshi The bug still exists in `1.5.0-beta01`
